Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2243,44 +2244,78 @@ extension BrowserViewController: UIAdaptivePresentationControllerDelegate {
 extension BrowserViewController {
     func presentIntroViewController(_ alwaysShow: Bool = false) {
         if showLoadingScreen(for: .shared) {
-            present(LoadingScreen(profile: profile, referrals: referrals, referralCode: User.shared.referrals.pendingClaim), animated: true)
+            presentLoadingScreen()
         } else if User.shared.firstTime {
-            User.shared.firstTime = false
-            User.shared.migrated = true
-            User.shared.hideRebrandIntro()
-            // deactivate searchbar hint for new users
-            contextHintVC.viewModel.markContextualHintPresented()
+            handleFirstTimeUserActions()
         } else if User.shared.showsRebrandIntro {
-            let intro = NTPIntroViewController()
-            intro.modalPresentationStyle = .overFullScreen
-            intro.modalTransitionStyle = .crossDissolve
-            present(intro, animated: true)
-            User.shared.hideRebrandIntro()
+            presentRebrandIntro()
+        } else {
+            presentInsightfulSheetsIfNeeded()
         }
-        presentDefaultBrowserPromoIfNeeded()
+    }
+
+    private func presentLoadingScreen() {
+        present(LoadingScreen(profile: profile, referrals: referrals, referralCode: User.shared.referrals.pendingClaim), animated: true)
+    }
+
+    private func handleFirstTimeUserActions() {
+        User.shared.firstTime = false
+        User.shared.migrated = true
+        User.shared.hideRebrandIntro()
+        // deactivate searchbar hint for new users
+        contextHintVC.viewModel.markContextualHintPresented()
+    }
+
+    private func presentRebrandIntro() {
+        let intro = NTPIntroViewController()
+        intro.modalPresentationStyle = .overFullScreen
+        intro.modalTransitionStyle = .crossDissolve
+        present(intro, animated: true)
+        User.shared.hideRebrandIntro()
     }
 
     private func showLoadingScreen(for user: User) -> Bool {
         (user.migrated != true && !user.firstTime)
-            || user.referrals.pendingClaim != nil
+                || user.referrals.pendingClaim != nil
     }
 
-    func presentDefaultBrowserPromoIfNeeded() {
-        let isHome = tabManager.selectedTab?.url.flatMap { InternalURL($0)?.isAboutHomeURL } ?? false
+    func presentInsightfulSheetsIfNeeded() {
+        if isHomePage() && presentedViewController == nil && !showLoadingScreen(for: .shared) {
+            if !presentDefaultBrowserPromoIfNeeded() {
+                presentWhatsNewPageIfNeeded()
```

Comment:
Wouldn't the whats new page have priority over the default browser promo?

Just thinking that whats new is more time pressing that default promo (which would not make a big difference to be shown next time).

Reply:
I personally agree with that 😅 - though I didn't want to modify the default behavior of the overall structure.
@ecotopian WDYT? It's an easy swap after all

Reply:
I guess it's a flip coin situation. Both is important. DefaultBrowser is historically the main lever to improve and measure retention. So I'd keep this as a precedence, until we have data to prove the opposite.